File: web/src/shared/api/compliance.ts
import { get, post, put } from './client';

// 一条管理员合规确认记录。
export interface ComplianceAck {
  version: string;
  admin_user_id: number;
  language: string;
  accepted_at: string;
}

// 合规确认状态（GET /admin/compliance 返回）。
export interface ComplianceStatus {
  required: boolean; // 是否仍需确认
  enabled: boolean; // 合规门是否开启
  acknowledged: boolean; // 是否已确认当前版本
  version: string;
  language: string;
  phrase: string; // 当前语言的逐字确认短语
  document_zh: string; // 中文合规文档全文（markdown）
  document_en: string; // 英文合规文档全文（markdown）
  ack?: ComplianceAck;
}

export const complianceApi = {
  // 查询当前管理员的合规确认状态（含文档与短语）。
  status: (language?: string) =>
    get<ComplianceStatus>('/api/v1/admin/compliance', language ? { language } : undefined),
  // 提交确认（逐字短语）。
  accept: (phrase: string, language: string) =>
    post<ComplianceStatus>('/api/v1/admin/compliance/accept', { phrase, language }),
  // 开启/关闭合规门。
  setEnabled: (enabled: boolean) =>
    put<ComplianceStatus>('/api/v1/admin/compliance/enable', { enabled }),
};

File: backend/internal/server/dto/compliance.go
package dto

// ===== 请求 =====

// ComplianceAcceptReq 提交管理员合规确认。
type ComplianceAcceptReq struct {
	// Phrase 须与当前语言要求的确认短语逐字匹配。
	Phrase string `json:"phrase" binding:"required"`
	// Language zh/en，缺省 zh。
	Language string `json:"language"`
}

// ComplianceEnableReq 开启/关闭合规门。
type ComplianceEnableReq struct {
	Enabled bool `json:"enabled"`
}

// ===== 响应 =====

// ComplianceAckResp 一条确认记录。
type ComplianceAckResp struct {
	Version     string `json:"version"`
	AdminUserID int    `json:"admin_user_id"`
	Language    string `json:"language"`
	AcceptedAt  string `json:"accepted_at"`
}

// ComplianceStatusResp 合规确认状态。
type ComplianceStatusResp struct {
	Required     bool   `json:"required"`     // 是否仍需确认
	Enabled      bool   `json:"enabled"`      // 合规门是否开启
	Acknowledged bool   `json:"acknowledged"` // 是否已确认当前版本
	Version      string `json:"version"`
	Language     string `json:"language"`
	Phrase       string `json:"phrase"`      // 当前语言确认短语
	DocumentZh   string `json:"document_zh"` // 中文合规文档全文（markdown）
	DocumentEn   string `json:"document_en"` // 英文合规文档全文（markdown）

	Ack *ComplianceAckResp `json:"ack,omitempty"` // 已确认时附带
}
